server/modules/protocol/MySQL/mysql_common.cc

```diff
@@ -1459,10 +1461,7 @@ int gw_decode_mysql_server_handshake(MySQLProtocol *conn, uint8_t *payload)
 
     mysql_server_capabilities_two = gw_mysql_get_byte2(payload);
 
-    memcpy(capab_ptr, &mysql_server_capabilities_one, 2);
-
-    // get capabilities part 2 (2 bytes)
-    memcpy(&capab_ptr[2], &mysql_server_capabilities_two, 2);
+    conn->server_capabilities = mysql_server_capabilities_one | mysql_server_capabilities_two << 16;
```

Comment:
Although this is better than the original, we try to avoid doing unrelated changes when they are not required. Please remove this change and submit it in a separate commit and pull request.

Reply:
server_capabilities is needed in function mxs_mysql_get_session_track_info to determine whether the server support extended ok packet;

Reply:
Ah, this was my mistake. I didn't realize that the `capab_ptr` was not used for anything. This looks good now.